### DISCUSSION TITLE
Ordering blocks

### DISCUSSION THREAD

**Discussion by damsfx** [wintercms/wn-blocks-plugin#12] [Ideas]
I was wondering what you might think about the possibility of ordering blocks in the list provided when a block is added?

Offering a predefined order can make end-users more productive.

Suggestion:
-Have a controller in the plugin settings that allows us to reorder the defined blocks.

When to use the setting:
-Use the sorting order ONLY if setting exists and synchronised with the available blocks.
or
-Use the sorting order if setting exists and add unordered or new blocks to the end of the list keeping the default plugin order.

Additional suggestions:
- Registed blocks can be enabled (default) or disabled, this setting may be reserved to authorized users with permission.

Thought and ideas welcome.

**Comment by LukeTowers** [comment]
Currently it's sorted alphabetically, but I think there's definitely some better UX we can have in general for the block editor piece, I'm not necessarily sure that ordering that list differently is going to make much of a difference but I do agree that I want something better than the grouped repeater interface.

**Reply by damsfx** [reply]
> I do agree that I want something better than the grouped repeater interface.

The repeated fields widget is pretty powerful, and can do some nice things, but it's true that it's a bit outdated.   
Are you thinking of a new form widget? ??

**Reply by LukeTowers** [reply]
Ideally I'd like something like a customized version of editor.js to act as the `blocks` formwidget instead of a grouped repeater.

**Comment by helmutkaufmann** [comment]
May I assume that there will no such option for the time being?

**Reply by LukeTowers** [reply]
None being worked on at the moment, you could always submit one yourself or sponsor the feature if you'd like 😉
